### ISSUE TITLE
Is there a way to output scss from scss?

### ISSUE THREAD

**Issue by thisischrisj** [sass/sass#2837]
It's going to sound like a strange question but I can't find anything online so bear with me. I'm wondering if there is a way to compile SCSS files into a sort of precompiled SCSS file that can be used by multiple sources.

Over the years, I've built up a bit of a complicated SCSS system to speed up my workflow. Historically, I would have a grunt or gulp task that just compiled all my SCSS to a single flat CSS file. With frameworks like Vue / React, I now split out any non-output utilities (variables, functions, mixins, silent classes) into their own "toolkit" and import that toolkit into my main CSS output as well as every component that might want to reference a variable, call a function, extend a silent class, etc... In these frameworks, any heavy SCSS lifting needs to be done by every component that imports the toolkit and this slows down compilation a lot.

As a hypothetical example, let's say I have a toolkit file that looks like this (I don't but run with me):

```
@for i from 1 to 1000000 {
    %p--#{i} {
        padding: i * 1px;
    }
}
```

And, for every Vue component (let's say there is 50 of them), I then have an 
`@import "toolkit"`
statement. When the project is running, any change to any file in the toolkit means all 50 components have to run that 1000000 loop. Every new component added to the projects adds another 1000000 loop and total compilation gets slower and slower.

I'm wondering if there's a way to precompile the toolkit so that I have a new file that looks like this:

```
%p--1 {
    padding: 1px;
}
%p--2 {
    padding: 2px;
}
%p--3 {
    padding: 3px;
}

...

%p--1000000 {
    padding: 1000000px;
}
```

Then, in every component, I can just:
`@import "compiled-toolkit"`
and then use statements such as:

```
.my-module {
    @extend %p--20;
}
```

**Comment by mirisuzanne** [comment]
If the main concern here is the performance of re-processing the same files for each component, I think this would be resolved by #2745.

That seems more likely to happen – and more directly to-the-point – than an output-scss-to-scss feature.

**Comment by thisischrisj** [comment]
Agreed and definitely a better solution. Thanks for pointing this out.
